lib/std/heap.zig

```diff
@@ -15,48 +15,88 @@ pub const ArenaAllocator = @import("heap/arena_allocator.zig").ArenaAllocator;
 
 const Allocator = mem.Allocator;
 
-pub const c_allocator = &c_allocator_state;
+usingnamespace if (comptime @hasDecl(c, "malloc_size")) struct {
+    pub const supports_malloc_size = true;
+    pub const malloc_size = c.malloc_size;
+} else if (comptime @hasDecl(c, "malloc_usable_size")) struct {
+    pub const supports_malloc_size = true;
+    pub const malloc_size = c.malloc_usable_size;
+} else struct {
+    pub const supports_malloc_size = false;
+};
+
+pub const c_allocator = mem.getAllocatorPtr(&c_allocator_state);
```

Comment:
The reason for `getAllocatorPtr` is so that `c_allocator_state` can be an `Allocator` or a struct wrapping an `Allocator` and this will still work. I was switching back and forth between wrapping `c_allocator_state` with `sanityWrap`.

My first instinct was to support a new method on all allocator types `fn allocatorRef(self: *Self) *Allocator`, including `mem.Allocator`, but I wanted to touch as little code as possible, so I went with adding `getAllocatorPtr`.  If this doesn't convince you to keep it them let me know and I will remove it.

Reply:
I think I would still prefer to remove it. I agree it's annoying to have to keep switching back and forth, and you had to do it a bunch of times in a bunch of different branches. Consider though, this code will be glanced over by thousands of people over the course of years, and for many of those the difference between having to follow a function call or see a field and look no further could be the difference between feeling like they fully understand the declaration, or thinking it's too complicated to bother reading the source. I know that sounds a bit like weird reasoning, I hope you can humor me on this one.

Reply:
That reasoning makes sense.  Thanks for the explanation, will change.